### PR TITLE
Create a form to add groups to room

### PR DESCRIPTION
**Pull request by Leonils**
This form display the user's groups and the user
can add them to a select list. Then when he submits
the groups ids, they are added to the room

Screenshot from 2022-07-06 16-05-01

### CODE REVIEW

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            onClick={onClose}
+            onFocus={(event: React.FocusEvent) => event.stopPropagation()}
+            margin={{ right: 'small' }}
+          />
+          <LoadingButton
+            label={intl.formatMessage(messages.addGroupsLabel, { numberSelected: selected.length })}
+            primary
+            onClick={() =>
+              mutate({ roomSlug, groupIds: selected.map((index) => groups?.[index]?.id as string) })
+            }
```
Comment (kernicPanel) [inline src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:137]: I think you should put this in a function.

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  return (
+    <Button
+      key={group.id}
+      onClick={(event: React.MouseEvent) => {
+        event.preventDefault();
+        event.stopPropagation();
+        onRemove(group.id);
+      }}
+      href="#"
+      onFocus={(event: React.FocusEvent) => event.stopPropagation()}
```
Comment (kernicPanel) [inline src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:46]: Several questions here. Why do we need:
- to stop the click and focus event propagation
- to specify this href

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            value={
+              <Box wrap direction="row">
+                {selected && selected.length > 0 ? (
+                  selected.map((groupIndex) => renderGroup(groups?.[groupIndex], handleUnselect))
+                ) : (
+                  <Box pad="small">{intl.formatMessage(messages.selectGroupsToAdd)}</Box>
+                )}
+              </Box>
+            }
+            options={(groups || []).map((g) => g.name)}
```
Comment (kernicPanel) [inline src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:118]:
```suggestion
            options={(groups || []).map((group) => group.name)}
```
Please don't use such shortened variable names.

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx (hunk 1 above)
Comment (Leonils) [reply src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:137]: Indeed.

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx (hunk 2 above)
Comment (Leonils) [reply src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:46]: This is the suggestion from grommet. 

As I understand it, we need to capture these events to stop the propagation to the select component (that would close the selector)

If we don't specify the href, it's considered a "button", and a "button" shouldn't be child of another button, but the select component of grommet is a button. Setting the "href" make it a link, and then it's valid.

You can see the example from grommet [here](https://storybook.grommet.io/?path=/story/input-select-children--children)

Would you have an alternative ?

File: src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx (hunk 3 above)
Comment (Leonils) [reply src/frontend/magnify/src/components/rooms/AddGroupToRoomDialog/AddGroupToRoomDialog.tsx:118]: Fair enough

**Review by kernicPanel — APPROVED** [review]
:+1:

**Pull request merged**